src/devices/Pcd8544/Pcd8544.cs

```diff
@@ -0,0 +1,487 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.Device.Pwm;
+using System.Threading;
+using System.Drawing;
+using Iot.Device.Display.Pcd8544Enums;
+
+namespace Iot.Device.Display
+{
+    /// <summary>
+    /// PCD8544 - 48 × 84 pixels matrix LCD
+    /// </summary>
+    public class Pcd8544 : IDisposable
+    {
+        /// <summary>
+        /// Size of the screen 48 x 84 / 8
+        /// </summary>
+        public const int ScreenSizeBytes = 504;
+
+        /// <summary>
+        /// The size of the screen in terms of characters
+        /// </summary>
+        public static Size Size => new Size(84, 6);
+
+        /// <summary>
+        /// The size of the screen in terms of characters
+        /// </summary>
+        public static Size PixelScreenSize => new Size(84, 48);
+
+        /// <summary>
+        /// Number of color per pixel
+        /// </summary>
+        public const int ColorPerPixel = 1;
+
+        private int _dataCommandPin;
+        private int _resetPin;
+        private PwmChannel? _pwmBacklight;
+        private SpiDevice _spiDevice;
+        private GpioController _controller;
+        private bool _shouldDispose;
+        private float _backlightVal = 0;
+        private bool _invd = false;
+        private byte _contrast = 0;
+        private int _position;
+
+        private byte[] _byteMap = new byte[504];
+        private byte _bias;
+        private bool _enabled;
+        private Temperature _temperature;
+
+        /// <summary>
+        /// Create Pcd8544
+        /// </summary>
+        /// <param name="dataCommandPin">The data command pin.</param>
+        /// <param name="resetPin">The reset pin. Use a negative number if you don't want to use it</param>
+        /// <param name="spiDevice">The SPI device.</param>
+        /// <param name="pwmBacklight">The PWM channel for the back light</param>
+        /// <param name="gpioController">The GPIO Controller.</param>
+        /// <param name="shouldDispose">True to dispose the GPIO controller</param>
+        public Pcd8544(int dataCommandPin, int resetPin, SpiDevice spiDevice, PwmChannel? pwmBacklight = null, GpioController? gpioController = null, bool shouldDispose = true)
+        {
+            if (dataCommandPin < 0)
+            {
+                throw new ArgumentException($"{nameof(dataCommandPin)} must be a valid pin number");
+            }
+
+            _dataCommandPin = dataCommandPin;
+            _pwmBacklight = pwmBacklight;
+            _pwmBacklight?.Start();
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+            _shouldDispose = gpioController == null || shouldDispose;
+            _controller = gpioController ?? new();
+            _resetPin = resetPin;
+            if (resetPin > 0)
```

Comment:
if reset pin is not required then I suggest to make it default to -1 and also check directly for -1 (I think pin 0 is perfectly valid number and you block it here)

Reply:
Yes, will do as well. That's actually good idea.

Reply:
And yes for >=0